### DISCUSSION TITLE
Questions about CAVSPHnrc "Cavity Zones" and "Material" and "Source" Input

### DISCUSSION THREAD

**Discussion by karenrex1023** [nrc-cnrc/EGSnrc#1202] [Questions] [answered]
Hello, I am trying to get a Kwall correction factor for a spherical ion chamber that I've modelled as concentric spheres (innermost sphere = C552 material to represent collector, middle sphere = air, outer sphere = wall). I was confused about the cavity zones input. When I change that input from 2 to 3, I get a different value for Kwall. I was wondering if you could clarify how I should specify this parameter. 
I also had a question on whether the transport medium before the spherical chamber geometry would be automatically set to air or vacuum in the simulation?

Finally, is there a way to specify the source in CAVSPHnrc as an isotropic cylindrical source rather than just a point source or a parallel beam? 
 
This is what I have as my input:
:start geometrical inputs:

 NUMBER OF CONES= 0        

 NUMBER OF SPHERES= 2      

 RADII=    0.2224, 0.945, 1.196 

 CAVITY ZONES= 3

 ######## Material Input 
 
MEDIA=              c552_512icru,
                     air512icru
 MEDNUM=        1, 2, 1 
 START REGION=  2,  3, 4
 STOP REGION=   2,  3, 4

 :stop geometrical inputs:

**Comment by rtownson** [comment, marked as answer]
You could consider switching to [cavity](https://nrc-cnrc.github.io/EGSnrc/doc/pirs898/cavity.html) or egs_chamber for more geometry and source flexibility.

The cavity zones are all the regions that comprise the cavity sensitive volume. See the region numbering scheme in section [7.2.1](https://nrc-cnrc.github.io/EGSnrc/doc/pirs702-egsnrc-codes.pdf). The region outside the sphere is region 1, so you can set its material just like any other region.
